provider/cldr/src/transform/list/cldr_serde.rs

```diff
@@ -0,0 +1,58 @@
+use crate::cldr_langid::CldrLangID;
+use serde::Deserialize;
+
+pub mod list_patterns_json {
+    //! Serde structs representing CLDR JSON numbers.json files.
+    //!
+    //! Sample file:
+    //! https://github.com/unicode-org/cldr-json/blob/main/cldr-json/cldr-misc-full/main/en/listPatterns.json
+
+    use super::*;
+
+    #[derive(PartialEq, Debug, Deserialize)]
+    pub struct ListPattern {
+        pub start: String,
+        pub middle: String,
+        pub end: String,
+        #[serde(rename = "2")]
+        pub pair: String,
+    }
+
+    #[derive(PartialEq, Debug, Deserialize)]
+    pub struct ListPatterns {
+        #[serde(rename = "listPattern-type-standard")]
+        pub standard: ListPattern,
+        #[serde(rename = "listPattern-type-standard-narrow")]
+        pub standard_narrow: ListPattern,
+        #[serde(rename = "listPattern-type-standard-short")]
+        pub standard_short: ListPattern,
+        #[serde(rename = "listPattern-type-or")]
+        pub or: ListPattern,
+        #[serde(rename = "listPattern-type-or-narrow")]
+        pub or_narrow: ListPattern,
+        #[serde(rename = "listPattern-type-or-short")]
+        pub or_short: ListPattern,
+        #[serde(rename = "listPattern-type-unit")]
+        pub unit: ListPattern,
+        #[serde(rename = "listPattern-type-unit-narrow")]
+        pub unit_narrow: ListPattern,
+        #[serde(rename = "listPattern-type-unit-short")]
+        pub unit_short: ListPattern,
```

Comment:
`question` this seems pretty hardcoded. It may benefit from more nested structure like:
```rust
    patterns: {
        or: Option(Enum(Single|Variants(Narrow, Short))),
        standard: Option(...),
        unit: Option(...)
    }
```

Reply:
This is the structure of the CLDR JSON files which this is parsing.

Reply:
It is a flatten one, because there are three types and three lengths per each. So you flatten it into a single struct with fields. I think it's okay, since if CLDR will add a new length or type you can keep extending this field list, but I'd prefer a more nested one.

Up to you. Non blocking.

Reply:
I don't think serde will let us bulk-rename on flatten; so I suspect this is written the best it can be, though I do agree that nesting+`#[serde(flatten)]` would be very nice if achievable